pybossa/view/projects.py

```diff
@@ -517,6 +518,54 @@ def clone(short_name):
         project=project_sanitized
     ))
 
+@blueprint.route('/<short_name>/tasks/taskpresenterimageupload', methods=['GET', 'POST'])
+@login_required
+@admin_or_subadmin_required
+@csrf.exempt
+def upload_task_guidelines_image(short_name):
+    errors = False
+    project = project_by_shortname(short_name)
+
+    disable_editor = (not current_user.admin and
+                      current_app.config.get(
+                          'DISABLE_TASK_PRESENTER_EDITOR'))
+
+    is_admin_or_owner = (
+        current_user.admin or
+        (project.owner_id == current_user.id or
+            current_user.id in project.owners_ids))
+
+    if disable_editor:
+        flash(gettext('Task presenter editor disabled!'), 'error')
+        errors = True
+    elif not is_admin_or_owner:
+        flash(gettext('Ooops! Only project owners can upload files.'), 'error')
+        errors = True
+
+    imgurls = []
+    for file in request.files.getlist("image"):
+        file_size_mb = file.seek(0, os.SEEK_END) / 1024 / 1024
+        file.seek(0, os.SEEK_SET)
+        file.filename = secure_filename(file.filename)
+        if file_size_mb < 5:
+            container = "user_%s" % current_user.id
+            uploader.upload_file(file, container=container)
+            imgurls.append(get_avatar_url(
+                current_app.config.get('UPLOAD_METHOD'),
+                file.filename,
+                container,
+                current_app.config.get('AVATAR_ABSOLUTE')
+            ))
+        else:
+            flash(gettext('File must be smaller than 5 MB.'))
+            errors = True
```

Comment:
If `errors` is a single boolean, should we rename this to just `error`?

Reply:
`errors` gets switched to `True` if the user is not an admin or owner, the editor is disabled, or if the files sizes are too big. So I think it makes sense to keep it plural.

Reply:
Understood. I suppose the concern is that plural `errors` indicates the value may be an array, rather than a boolean. If the value will always be a single value of True or False, perhaps `error` would be reasonable to avoid confusion of the data-type?

Reply:
Makes sense, Changing to `error`